src/Hint/Restrict.hs

```diff
@@ -157,6 +158,11 @@ checkPragmas modu flags exts mps =
      , not $ null bad]
    isGood def mp x = maybe def (within modu "" . riWithin) $ Map.lookup x mp
 
+
+-- | Extension to GHC's 'ImportDeclQualifiedStyle', expressing @qualifiedStyle: unrestricted@,
+-- i.e. the preference of "either pre- or post-, but qualified" in a rule.
+data QualifiedPostOrPre = QualifiedPostOrPre deriving Eq
```

Comment:
I chose a singleton datatype instead of another Maybe layer, basically to have `Either QualifiedPostOrPre ImportDeclQualifiedStyle` sum type with values for:
 * QualifiedPostOrPre,
 * QualifiedPre, 
 * QualifiedPost,
 * NotQualified.

`checkImports` already uses 2 or 3 layers of Maybe, and I think it was getting confusing.

Reply:
Seems like it might even be better to make a new data type like `data ImportStyle = PostOrPre | ??? ImportDeclQualifiedStyle` if you could figure out a good name for `???`